### DISCUSSION TITLE
Accessing the map view from the controller

### DISCUSSION THREAD

**Discussion by mlandsfeld** [tethysplatform/tethys#774] [Q&A]
Is there a way to access the current Openlayers map view from the python controller? I would like to maintain the current map view when re-rendering the web page by sending the current view back in the context. Thanks!

**Comment by swainn** [comment]
You cannot access the current open layers map view that is running in the browser from the controller. However, there are several ways to accomplish what you are looking for. 

1. One way would be to use AJAX calls in JavaScript to send data back to your app and sync certain settings. For example, you could create database tables for storing the settings you'd like to persist when the page reloads. Then create a REST endpoint in your app that can be called to update the values of the settings. Any time one of these changes in the map, use JavaScript to send an AJAX request on the REST endpoint to send the updates to the server. This could also be accomplished using WebSockets, which allow 2-way communication between the browser and server. The [THREDDS](http://docs.tethysplatform.org/en/stable/tutorials/thredds.html) and [Google Earth Engine](http://docs.tethysplatform.org/en/stable/tutorials/google_earth_engine.html) tutorials would be good resources for learning AJAX and setting up REST endpoints in your app. The [WebSockets Concepts](http://docs.tethysplatform.org/en/stable/tutorials/websockets.html) tutorial could get you started on WebSockets. 

2. If you want to persist settings for each user, one way would be to save them in "local storage" which is a key-value database in the web browser. Local storage is browser-specific, so if they opened the application up on a different device, they likely wouldn't see the saved changes. It also wouldn't work to store changes to settings that are meant to be global (changed for all users). See: [Window.localStorage](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage)
